pkg/controller/controller_binding.go

```diff
@@ -581,6 +625,18 @@ func (c *controller) reconcileServiceInstanceCredential(binding *v1alpha1.Servic
 		glog.V(5).Infof("Successfully deleted ServiceInstanceCredential %v/%v of ServiceInstance %v/%v of ServiceClass %v at ServiceBroker %v", binding.Namespace, binding.Name, instance.Namespace, instance.Name, serviceClass.Name, brokerName)
 	}
 
+	if toUpdate.Status.OrphanMitigationInProgress {
+		s := fmt.Sprintf(`Orphan mitigation successful for ServiceInstanceCredential "%v/%v", setting condition to failed`, binding.Namespace, binding.Name)
+		c.setServiceInstanceCredentialCondition(toUpdate,
+			v1alpha1.ServiceInstanceCredentialConditionFailed,
+			v1alpha1.ConditionTrue,
+			"OrphanMitigationSuccessful",
+			s)
+		if _, err := c.updateServiceInstanceCredentialStatus(toUpdate); err != nil {
+			return err
+		}
+		toUpdate.Status.OrphanMitigationInProgress = false
```

Comment:
I'd add `toUpdate.Status.OrphanMitigationInProgress = false` to the `clearServiceInstancfeCredentialCurrentOperation` function, as I would consider orphan mitigation done if the deprovision hits a failed state.

Reply:
Wouldn't doing that mean orphan mitigation was only attempted once?

Reply:
The current operation is only cleared when the operation has completely finished. So in this case, once the controller has either successfully removed the binding or has given up on removing it (either due to an HTTP error or a timeout). In the case of other errors, the CurrentOperation isn't cleared, so the controller would still keep trying to do the orphan mitigation.

Reply:
From the spec:

```
To mitigate orphan service instances and bindings, the marketplace SHOULD attempt to delete resources it cannot be sure were successfully created, and SHOULD keep trying to delete them until the broker responds with a success.
```